src/main/java/nextstep/subway/applicaion/LineService.java

```diff
@@ -181,10 +172,67 @@ public void deleteSection(Long lineId, Long stationId) {
         Line line = lineRepository.findById(lineId).orElseThrow(IllegalArgumentException::new);
         Station station = stationService.findById(stationId);
 
-        if (!line.getSections().get(line.getSections().size() - 1).getDownStation().equals(station)) {
+        validateSectionDeleteRequest(line, station);
+
+        List<Section> sections = line.getSections();
+        List<Section> filteredSections = sections.stream()
+                .filter(
+                        s -> (s.getUpStationId() == stationId) || (s.getDownStationId() == stationId)
```

Comment:
필터 내 조건문이 복잡해지는 경우 그 의미를 명확히 하기 위해 메서드로 분리 후 
메서드 이름으로 그 목적을 나타내면 조금 더 가독성이 좋아질 수 있습니다.

```suggestion
                        s -> s.checkBelongingStationId(stationId)
```

Reply:
@boorownie 
다음 단계에서 pr에서 반영해서 올리도록 하겠습니다!

Reply:
반영했습니다! (https://github.com/next-step/atdd-subway-path/pull/641/commits/998df63af4bd5603cca03e6d5d70da1d24f40ae4)